components/brave_search/renderer/brave_search_default_js_handler.cc

```diff
@@ -0,0 +1,157 @@
+      "setIsDefaultSearchProvider",
+      base::BindRepeating(
+          &BraveSearchDefaultJSHandler::SetIsDefaultSearchProvider,
+              base::Unretained(this), isolate));
+}
+
+template <typename Sig>
+void BraveSearchDefaultJSHandler::BindFunctionToObject(
+    v8::Isolate* isolate,
+    v8::Local<v8::Object> javascript_object,
+    const std::string& name,
+    const base::RepeatingCallback<Sig>& callback) {
+  v8::Local<v8::Context> context = isolate->GetCurrentContext();
+  // Get the isolate associated with this object.
+  javascript_object
+      ->Set(context, gin::StringToSymbol(isolate, name),
+            gin::CreateFunctionTemplate(isolate, callback)
+                ->GetFunction(context)
+                .ToLocalChecked())
+      .Check();
+}
+
+v8::Local<v8::Promise> BraveSearchDefaultJSHandler::IsDefaultSearchProvider(
+    v8::Isolate* isolate) {
+  if (!EnsureConnected())
+    return v8::Local<v8::Promise>();
+
+  v8::MaybeLocal<v8::Promise::Resolver> resolver =
+      v8::Promise::Resolver::New(isolate->GetCurrentContext());
+  if (!resolver.IsEmpty()) {
+    auto promise_resolver =
+        std::make_unique<v8::Global<v8::Promise::Resolver>>();
+    promise_resolver->Reset(isolate, resolver.ToLocalChecked());
+    auto context_old = std::make_unique<v8::Global<v8::Context>>(
+        isolate, isolate->GetCurrentContext());
+    brave_search_default_->IsDefaultSearchProvider(
+        base::BindOnce(&BraveSearchDefaultJSHandler::OnIsDefaultSearchProvider,
+                       base::Unretained(this), std::move(promise_resolver),
+                       isolate, std::move(context_old)));
+
+    return resolver.ToLocalChecked()->GetPromise();
+  }
+
+  return v8::Local<v8::Promise>();
+}
+
+void BraveSearchDefaultJSHandler::OnIsDefaultSearchProvider(
+    std::unique_ptr<v8::Global<v8::Promise::Resolver>> promise_resolver,
+    v8::Isolate* isolate,
+    std::unique_ptr<v8::Global<v8::Context>> context_old,
+    const bool response) {
+  v8::HandleScope handle_scope(isolate);
+  v8::Local<v8::Context> context = context_old->Get(isolate);
+  v8::Context::Scope context_scope(context);
+
+  v8::Local<v8::Promise::Resolver> resolver = promise_resolver->Get(isolate);
+  v8::Local<v8::Boolean> result;
+  result = v8::Boolean::New(isolate, response);
+
+  ALLOW_UNUSED_LOCAL(resolver->Resolve(context, result));
+}
+
+void BraveSearchDefaultJSHandler::SetIsDefaultSearchProvider(
+    v8::Isolate* isolate) {
+  if (!EnsureConnected())
+    return;
+  // TODO(petemill): only for user activation
+  // LocalFrame::HasTransiendUserActivation
+  auto* web_frame = render_frame_->GetWebFrame();
+  // Prevent site from calling this in response to a DOM event or Timer.
+  if (web_frame->HasTransientUserActivation()) {
+    // TODO(petemill): Consider showing a confirmation dialog to user.
+    brave_search_default_->SetIsDefaultSearchProvider();
+  } else {
+    blink::WebString message = "setIsDefaultSearchProvider: "
```

Comment:
seems like this should return an error to the page?

Reply:
Well, this is to block malicious or incorrect use of the script i.e. not putting it in a click handler. So logging an error in devtools is ok because the devs will know at dev-time that they haven't put it in the right place. Seems like this is how it's done in other places in chromium 
/ blink.